### PR TITLE
Iddes les indicator

### PR DESCRIPTION
**Pull request by psakievich**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [s] Feature enhancement

## Description of the pull-request

Provide a field for checking LES/RANS regions in the iddes model.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [ ] Linux
    - [x] MacOS
  - Compilers 
    - [ ] GCC
    - [x] LLVM/Clang
    - [ ] Intel compilers
    - [ ] NVIDIA CUDA
- [x] Compiles without warnings
- [x] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

File: src/node_kernels/TKESSTIDDESNodeKernel.C [hunk 1]
```diff
@@ -47,6 +47,7 @@ TKESSTIDDESNodeKernel::setup(Realm& realm)
   dualNodalVolume_ = fieldMgr.get_field<double>(dualNodalVolumeID_);
   maxLenScale_     = fieldMgr.get_field<double>(maxLenScaleID_);
   fOneBlend_       = fieldMgr.get_field<double>(fOneBlendID_);
+  lengthScaleRatio_ = fieldMgr.get_field<double>(lengthScaleRatioID_);
```
Comment (sayerhs) [inline src/node_kernels/TKESSTIDDESNodeKernel.C:50]:
```suggestion
  lengthScaleRatio_ = fieldMgr.get_field<double>(lengthScaleRatioID_);
  // call modify before this field gets modified in kernel execute phase
  lengthScaleRatio_.modify_on_device();
```

File: src/node_kernels/TKESSTIDDESNodeKernel.C [hunk 2]
```diff
@@ -122,11 +123,14 @@ void TKESSTIDDESNodeKernel::execute(
...
   const DblType lSST = sqrtTke / betaStar_ / sdr;
+  const DblType lLES = cDES * delta;
 
   // Find minimum length scale, limit minimum value to 1.0e-16 to prevent
   // division by zero later on
   const DblType lIDDES =
-      stk::math::max(1.0e-16, fdHat * (1.0  + fe) * lSST + (1.0 - fdHat) * cDES * delta);
+    stk::math::max(1.0e-16, fdHat * (1.0 + fe) * lSST + (1.0 - fdHat) * lLES);
+
+  lengthScaleRatio_.get(node, 0) = lLES / lIDDES;
```
Comment (sayerhs) [inline src/node_kernels/TKESSTIDDESNodeKernel.C:133]:
```suggestion
  // modify on device called in setup
  lengthScaleRatio_.get(node, 0) = lLES / lIDDES;
```

**Review by sayerhs — COMMENTED** [review]
The field registration and population logic looks fine for CPU runs. However, there is an issue when running on GPUs... we never call `modify_on_device()` so the field might not get synced back to host before being written to exodus files. 

I added a few comments where we could sneak in `modify_on_device()`, but happens in setup away from where the field is actually modified... which is sub-optimal. You can either do that, or merge as is and open an issue so that we don't forget about this when executing on GPUs.

**Comment by psakievich** [comment]
Thanks @sayerhs good point.  I'll update the PR.  I like the idea of keeping it device compatible at all times.

**Pull request merged**